build_library/catalyst.sh

```diff
@@ -3,6 +3,11 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+# Before doing anything, ensure we have at least Catalyst 4.
+if catalyst --version | grep -q "Catalyst [0-3]\."; then
+    sudo emerge -v1 ">=dev-util/catalyst-4" || exit 1
```

Comment:
could use a `--jobs ${NUM_JOBS}`.

catalyst.sh is a library file and meant to be sourced, executing a build command from such a file is a bad idea. this would fit better in catalyst_init() or in the caller.

Reply:
Both good points. I was originally going to put it in `catalyst_init` but then thought I needed the upgrade to happen earlier. I was wrong though, so I'll do that.